### PR TITLE
feat(React): Avatar dropdown menu and logout function

### PR DESCRIPTION
**Pull request by brendansun93**
**Scope**
This PR changes all page that includes the header bar. 

**Changes**
This PR adds new feature which pops up a menu when hovering on profile on top right corner of the page. Currently, there is only 1 option, that is to log out. After this option is clicked, user session is cleared and users are redirected to the login page. 

_This PR also contains a overwritten changes in https://github.com/linkedin/datahub/pull/2095 where multiple clicking on the profile will redirect users to url like "user/user/user/user/..."_
Screenshot from 2021-02-17 16-39-54

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Comment by brendansun93** [comment]
@gabe-lyons @jjoyce0510 @shirshanka 
Please review

File: datahub-web-react/src/app/shared/ManageAccount.tsx [hunk 1]
```diff
@@ -16,17 +18,43 @@ const defaultProps = {
...
+                <div tabIndex={0} role="button" onClick={handleLogout} onKeyDown={handleLogout}>
+                    Log out
+                </div>
+            </Menu.Item>
+        </Menu>
+    );
+
+    if (!isLoggedIn) {
+        return <Redirect to="/login" />;
+    }
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/shared/ManageAccount.tsx:43]: do you need this redirect? Each page is wrapped in a ProtectedRoute component which should be doing this redirect for you

File: datahub-web-react/src/app/shared/ManageAccount.tsx [hunk 2]
```diff
@@ -16,17 +18,43 @@ const defaultProps = {
...
-                style={{
-                    marginRight: '15px',
-                    color: '#f56a00',
-                    backgroundColor: '#fde3cf',
-                }}
-                src={_pictureLink || defaultAvatar}
-            />
-        </Link>
+        <Dropdown overlay={menu}>
+            <Link to={`/${entityRegistry.getPathName(EntityType.CorpUser)}/${_urn}`}>
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/shared/ManageAccount.tsx:47]: nice re-fix 👍

**Review by gabe-lyons — APPROVED** [review]
looks great, thanks @brendansun. I do think you may be able to remove the redirect logic though.

File: datahub-web-react/src/app/shared/ManageAccount.tsx [hunk 3]
```diff
@@ -16,17 +18,43 @@ const defaultProps = {
 
 export const ManageAccount = ({ urn: _urn, pictureLink: _pictureLink }: Props) => {
     const entityRegistry = useEntityRegistry();
+
+    const isLoggedIn = checkAuthStatus();
+
+    const handleLogout = () => {
+        Cookies.remove('PLAY_SESSION');
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/shared/ManageAccount.tsx:25]: technically, this line isn't doing anything. the play cookie is http only, so js isn't able to read or write it

Comment (brendansun93) [reply datahub-web-react/src/app/shared/ManageAccount.tsx]: This line is removed now.

File: datahub-web-react/src/app/shared/ManageAccount.tsx (hunk 1 above)
Comment (brendansun93) [reply datahub-web-react/src/app/shared/ManageAccount.tsx]: You are right, this is removed now.

**Comment by brendansun93** [comment]
@gabe-lyons Comments resolved, please review again. Thanks!

**Review by gabe-lyons — APPROVED** [review]
perfect 🚀

**Review by shirshanka — APPROVED** [review]
LGTM! 
Although we never want people to log out of DataHub :)

**Comment by jjoyce0510** [comment]
Thanks Brendan!

**Pull request merged**